GitHubTeams.ps1

```diff
@@ -273,6 +321,482 @@ filter Get-GitHubTeamMember
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .INPUTS
+        GitHub.Organization
+        GitHub.Team
+
+    .OUTPUTS
+        GitHub.Team
+
+    .EXAMPLE
+        Set-GitHubTeam -OrganizationName PowerShell -TeamName Developers -Description 'New Description'
+
+        Updates the description for the 'Developers' GitHub team in the 'PowerShell' organization.
+
+    .EXAMPLE
+        $team = Get-GitHubTeam -OrganizationName PowerShell -TeamName Developers
+        $team | Set-GitHubTeam -Description 'New Description'
+
+        You can also pipe in a GitHub team that was returned from a previous command.
+#>
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '',
+        Justification = 'One or more parameters (like NoStatus) are only referenced by helper
+        methods which get access to it from the stack via Get-Variable -Scope 1.')]
+    [CmdletBinding(
+        SupportsShouldProcess,
+        PositionalBinding = $false
+    )]
+    [OutputType( { $script:GitHubTeamTypeName } )]
+    param
+    (
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            Position = 1)]
+        [ValidateNotNullOrEmpty()]
+        [string] $OrganizationName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            Position = 2)]
+        [ValidateNotNullOrEmpty()]
+        [string] $TeamName,
+
+        [string] $Description,
+
+        [ValidateSet('Secret','Closed')]
+        [string] $Privacy,
+
+        [string] $ParentTeamName,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $telemetryProperties = @{
+        OrganizationName = (Get-PiiSafeString -PlainText $OrganizationName)
+        TeamName = (Get-PiiSafeString -PlainText $TeamName)
+    }
+
+    $getGitHubTeamParms = @{
+        OrganizationName = $OrganizationName
+    }
+    if ($PSBoundParameters.ContainsKey('AccessToken'))
+    {
+        $getGitHubTeamParms['AccessToken'] = $AccessToken
+    }
+    if ($PSBoundParameters.ContainsKey('NoStatus'))
+    {
+        $getGitHubTeamParms['NoStatus'] = $NoStatus
+    }
+
+    $orgTeams = Get-GitHubTeam @getGitHubTeamParms
+
+    $team = $orgTeams | Where-Object -Property name -eq $TeamName
+
+    $uriFragment = "/orgs/$OrganizationName/teams/$($team.slug)"
+
+    $hashBody = @{
+        name = $TeamName
```

Comment:
This current design prohibits being able to rename a team, even though it's a scenario that the API permits.  This is primarily due to you using the TeamName as the identifier instead of the slug.  Was preventing a team rename an intentional design decision, or an oversight?  An additional parameter of `NewTeamName` (or similar) would unblock this.

Reply:
This would be better achieved with a `Rename-GitHubTeam` function, but that would need to be as another PR.